### PR TITLE
uploads: Add Bluesky support

### LINKED ISSUE #5640: Bluesky support
Post page URL regex: `^https://bsky.app/profile/(.+?)/post/(.+)$` (groups: HANDLE, POST_ID)
Thumbnail URL regex: `^https://cdn.bsky.app/img/.+?/plain/(.+?)/(.+?)@.*$` (groups: DID, CID)
Original Image URL regex: `^https://(?:bsky\.social|.*\.bsky\.network)/xrpc/com\.atproto\.sync\.getBlob\?did=(.*)&cid=(.*)$` (groups: DID, CID)

`DID` from `HANDLE` can be retrieved using [com.atproto.identity.resolveHandle](https://www.docs.bsky.app/docs/api/com-atproto-identity-resolve-handle) API call.

<details>
  <summary>Example:</summary>

  ```sh
  $ curl -L 'https://bsky.social/xrpc/com.atproto.identity.resolveHandle?handle=ixy.bsky.social' | jq .did
  "did:plc:3jogsxcisdcdzwjobhxbav2w"
  ```
</details>

`CID`s from post can be retrieved using [app.bsky.feed.getPostThread](https://www.docs.bsky.app/docs/api/app-bsky-feed-get-post-thread) API call.

<details>
  <summary>Example:</summary>
  
  ```sh
  $ curl -L 'https://bsky.social/xrpc/app.bsky.feed.getPostThread?uri=at://did:plc:3jogsxcisdcdzwjobhxbav2w/app.bsky.feed.post/3kkvo4d4jd32g' -H "Authorization: Bearer $JWT" | jq '.thread.post.record.embed.images[] | {cid: .image.ref."$link", mime: .image.mimeType}'
  {
    "cid": "bafkreiawa4vn5k37h2mlpwuhaqmeog3hsfe3z47iot7reqxjlff6juyge4",
    "mime": "image/jpeg"
  }
  ```
  Note: `$JWT` contains access jwt token. See details below.
</details>

Original image can be retrieved from `DID` and `CID` using [com.atproto.sync.getBlob](https://www.docs.bsky.app/docs/api/com-atproto-sync-get-blob) API call. File extension is not available, but can be deduced from returned mime type, or by using good old `libmagic`.

<details>
  <summary>Example:</summary>

  ```sh
  $ curl -L 'https://bsky.social/xrpc/com.atproto.sync.getBlob?did=did:plc:3jogsxcisdcdzwjobhxbav2w&cid=bafkreiawa4v
n5k37h2mlpwuhaqmeog3hsfe3z47iot7reqxjlff6juyge4' -o image.ext

  $ sha256sum image.ext 
  16072adeab7f3e98b7da870418471b679149bcf3e874ff1242e9594be4d30627  image.ext

  $ file --mime-type image.ext
  image.ext: image/jpeg
  ```
</details>

Auth token can be retrieved using [com.atproto.server.createSession](https://www.docs.bsky.app/docs/api/com-atproto-server-create-session) API call. [com.atproto.server.refreshSession](https://www.docs.bsky.app/docs/api/com-atproto-server-refresh-session
) to refresh expired access tokens.

### PR DESCRIPTION
**Pull request by hdk5**
Kinda draft state for now tbh.
The two very basic test cases (post url and direct thumb url) work, but there may be some extra stuff I didn't notice

Fixes #5640

### CODE REVIEW

**Comment by hdk5** [comment]
https://bsky.app/profile/yourbaguette.bsky.social/post/3kjarhifsmg26 is broken - reminder to fix later

**Comment by hdk5** [comment]
I think it is ready enough to be tested in prod

**Comment by TypeA2** [comment]
> https://bsky.app/profile/yourbaguette.bsky.social/post/3kjarhifsmg26 is broken - reminder to fix later

This one broke my script too, for some reason the images array is under the `media` key, I didn't dig into why that is though:
https://gist.github.com/TypeA2/dbe1125f4256a21f3cbc1a516a64a99a#file-bluesky_replacement-py-L154

File: app/logical/source/extractor/bluesky.rb [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    if response["error"].present?
+      DanbooruLogger.info("Bluesky login failed (#{response["message"]} #{response["message"]})")
+      nil
+    else
+      response["accessJwt"]
+    end
+  end
+
+  memoize def cached_access_token
+    Cache.get("bluesky-access-token", 1.hours, skip_nil: true) do
```
Comment (hdk5) [inline app/logical/source/extractor/bluesky.rb:133]: By the way, the access token is issued for 2 hours, the refresh token - for 90 days. The token is JWT, so its `iat` and `exp` can be checked as well, but I don't think it's worth bothering doing the intended auth flow.

**Pull request merged**